### ISSUE TITLE
feat(ml): expose kailash_ml.metrics public module with standalone metric functions

### ISSUE THREAD

**Issue by esperie** [terrene-foundation/kailash-py#341]
## Summary

kailash-ml computes metrics internally during `TrainingPipeline.train()` via `EvalSpec(metrics=["accuracy", "f1"])`, but there is **no public module** exposing standalone metric functions. Every ASCENT exercise that needs to evaluate predictions falls back to `from sklearn.metrics import ...`.

## Problem

```python
# This doesn't exist — forces raw sklearn imports
from kailash_ml.metrics import roc_auc_score, f1_score, silhouette_score
score = roc_auc_score(y_true, y_pred)
```

14 ASCENT solution files across M3 and M4 import `sklearn.metrics` directly because kailash-ml provides no alternative.

## Proposed API

```python
from kailash_ml.metrics import (
    # Classification
    accuracy_score, f1_score, precision_score, recall_score,
    roc_auc_score, average_precision_score, log_loss, brier_score_loss,
    classification_report, matthews_corrcoef, cohen_kappa_score,
    precision_recall_curve, roc_curve, confusion_matrix,
    # Regression
    mean_squared_error, root_mean_squared_error, mean_absolute_error,
    r2_score, explained_variance_score, mean_absolute_percentage_error,
    median_absolute_error, max_error,
    # Clustering
    silhouette_score, adjusted_rand_score, calinski_harabasz_score,
    davies_bouldin_score,
    # Ranking
    ndcg_score, top_k_accuracy_score,
)
```

These can be thin wrappers around sklearn.metrics (consistent with how kailash-ml already wraps sklearn internally), but exposed as first-class public API.

## Implementation Notes

- `_shared.compute_metrics_by_name` already handles 9 metrics internally — this module should reuse that logic and extend it
- The internal `_METRIC_MAP` in `_shared.py` (lines 82-160) is the natural place to add missing entries
- The public module should accept the same `(y_true, y_pred)` / `(y_true, y_score)` signatures as sklearn
- Clustering metrics take `(X, labels)` or `(labels_true, labels_pred)` — different signature family

## Priority

**P0** — This is the single most impactful gap. Fixing this eliminates raw sklearn imports from 10+ ASCENT exercise files.

## Affected ASCENT Exercises

- ascent03/solutions/ex_1 through ex_8 (7 files)
- ascent04/solutions/ex_1 through ex_5, ex_7 (6 files)

**Comment by esperie** [comment]
Resolved in 090c0e97 — feat(platform): full clearance
